src/fat/bpb.rs

```diff
@@ -85,23 +85,23 @@ impl<'a> Bpb<'a> {
     // FAT16/FAT32 functions
 
     /// Get the Volume Label string for this volume
-    pub fn volume_label(&self) -> &[u8] {
-        if self.fat_type != FatType::Fat32 {
-            &self.data[43..=53]
-        } else {
-            &self.data[71..=81]
+    pub fn volume_label(&self) -> [u8; 11] {
+        let mut result = [0u8; 11];
+        match self.fat_type {
+            FatType::Fat16 => result.copy_from_slice(&self.data[43..=53]),
+            FatType::Fat32 => result.copy_from_slice(&self.data[71..=81]),
```

Comment:
According to https://en.wikipedia.org/wiki/Design_of_the_FAT_file_system, this field is not available on Fat32 "if the signature at offset 0x042 is set to 0x28".

Perhaps that should be checked here?

Reply:
> Most FAT32 file system implementations do not support an alternative signature of 0x28

> Should be 0x29 to indicate that an EBPB with the following 3 entries exists (since OS/2 1.2 and DOS 4.0). Can be 0x28 on some OS/2 1.0-1.1 and PC DOS 3.4 disks indicating an earlier form of the EBPB format with only the serial number following

I'm OK with not supporting disks formatted with MS-DOS before 4.0.